providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -421,7 +424,7 @@ static int rsa_set_ctx_params(void *vprsactx, const OSSL_PARAM params[])
     char mdprops[OSSL_MAX_PROPQUERY_SIZE] = { '\0' };
     char *str = mdname;
 
-    if (prsactx == NULL || params == NULL)
+    if (prsactx == NULL)
```

Comment:
I think I prefer the pattern of
````
if (params == NULL)
    return 1;
````

otherwise it needs to do all the locates for no reason..

Reply:
Hmmm, this could be painful :)

The params == NULL case will fast exit from the locates, so the cost isn't huge.  I'll change some but I doubt I'll get all of them.

Reply:
Not worth doing if it only does one or 2 locates.. A pageful migbt be good to skip though,